build.psm1

```diff
@@ -286,122 +287,191 @@ function Start-PSBuild {
 "@
-                log "  Executing Build Command: $command"
-                Start-NativeExecution { Invoke-Expression -Command:$command }
+                    log "  Executing Build Command: $command"
+                    Start-NativeExecution { Invoke-Expression -Command:$command }
 
-                $clrTarget = "FullClr"
-                if ($OneCoreValue -eq "ON")
-                {
-                    $clrTarget = "CoreClr"
+                    $clrTarget = "FullClr"
+                    if ($OneCoreValue -eq "ON")
+                    {
+                        $clrTarget = "CoreClr"
+                    }
+
+                    # Copy the binaries from the local build directory to the packaging directory
+                    $dstPath = ($script:Options).Top
+                    $FilesToCopy | % {
+                        $srcPath = Join-Path (Join-Path (Join-Path (Get-Location) "bin") $msbuildConfiguration) "$clrTarget/$_"
+                        log "  Copying $srcPath to $dstPath"
+                        Copy-Item $srcPath $dstPath
+                    }
                 }
 
-                # Copy the binaries from the local build directory to the packaging directory
-                $dstPath = ($script:Options).Top
-                $FilesToCopy | % {
-                    $srcPath = Join-Path (Join-Path (Join-Path (Get-Location) "bin") $msbuildConfiguration) "$clrTarget/$_"
-                    log "  Copying $srcPath to $dstPath"
-                    Copy-Item $srcPath $dstPath
+                if ($FullCLR) {
+                    $fullBinaries = @(
+                        'powershell.exe',
+                        'powershell.pdb',
+                        'pwrshplugin.dll',
+                        'pwrshplugin.pdb'
+                    )
+                    Build-NativeWindowsBinaries "OFF" $fullBinaries
+                }
+                else
+                {
+                    $coreClrBinaries = @(
+                        'pwrshplugin.dll',
+                        'pwrshplugin.pdb'
+                    )
+                    Build-NativeWindowsBinaries "ON" $coreClrBinaries
                 }
+            } finally {
+                Pop-Location
             }
-
-            if ($FullCLR) {
-                $fullBinaries = @(
-                    'powershell.exe',
-                    'powershell.pdb',
-                    'pwrshplugin.dll',
-                    'pwrshplugin.pdb'
+        }
+    }
+    else
+    {
+        # use native binaries from $PSHOME, assume it is latest
+        $nativeBinaries = @()
+        if ($IsWindows)
+        {
+            if ($PSVersionTable.PSEdition -ne "Core")
+            {
+                # if running under Windows PowerShell, we'll need to get the binaries from GitHub release
```

Comment:
Here are my 2 cents:
1. I think we should by default continue to build native component for OSX and Linux. Unlike `pwrshplugin.dll`, `libpsl` is much more crucial to PSCore (it has important native implementations that PS engine needs to call in via P/inovke), and it changes far more frequently than the WinRM plugin dll -- for example, @jeffbi made changes to it to fix the cyclic symbolic link detection issue; @PaulHigin made changes to it to create the SSH process for remoting.
2. For `pwrshplugin.dll`, `pwrshplugin.pdb` and `Install-PowerShellRemoting.ps1`, they should be made into a NuGet package and published to powershell-core feed. So we can depend on dotnet-cli to handle them and it's also more robust and clean. When we need to make changes to `pwrshplugin.dll`, we just publish a new version of the NuGet package.